packages/medusa/src/repositories/customer.ts

```diff
@@ -3,10 +3,7 @@ import { Customer } from "../models/customer"
 
 @EntityRepository(Customer)
 export class CustomerRepository extends Repository<Customer> {
-  async listAndCount(query): Promise<[Customer[], number]> {
-    const groups = query?.where?.groups
-    delete query?.where?.groups
-
+  async listAndCount(query, groups): Promise<[Customer[], number]> {
```

Comment:
I really don't like that we have to add additional arguments to do "nested" queries, but I can see why it is necessary in this case - let's maybe investigate an alternative refactor for the future so that all of these "common" queries will be handled consistently across all repositories.